### PR TITLE
Fix issue where audio output is not updated if chosen before meeting starts

### PR DESCRIPTION
**Pull request by ltrung**
**Issue #2242:**
Calling `chooseAudioOutput` before calling `audioVideo.start` does not work because audio video controller only subscribes to `audioOutputChange` event during connection. This is changed in PR #2163.

**Description of changes:**
Subscribe to `audioOutputChange` in audio mix controller instead of audio video controller (note that audio video controller still handles audio and video input change events as before). This works as audio video controller does not actually handle anything with audio output and just call audio mix controller anyway.

**Testing:**

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
- Join a meeting in Chrome and select a non-default audio output in the preview screen
- When in a meeting, verify that you can hear audio from the one you selected before

**Checklist:**

1. Have you successfully run `npm run build:release` locally? Yes

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? No

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved? No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Replace `startVideoInput(null)` and `startAudioInput(null)` with`stopVideoInput` and `stopAudioInput` for video, audio test in meeting readiness checker to stop video, audio input.
 - Replace the deprecated API `getRTCPeerConnectionStats` with `metricsDidReceive` in meeting readiness checker. 
 - Prevent `realtimeUnsubscribeFromVolumeIndicator` from causing a fatal error when there are no subscriptions for the `attendeeId`.
+- Subscribe to `audioOutputDidChange` in audio mix controller to fix the issue where the audio output is not updated 
+  before meeting start.
```
Comment (xuesichao) [inline CHANGELOG.md:21]: nit: better not to split into two lines

**Review by xuesichao — APPROVED** [review]
Looks good to me. It's good to have test steps in the description section. But I guess it's hard to test and maybe not necessary.

File: src/audiovideocontroller/DefaultAudioVideoController.ts [hunk 2]
```diff
@@ -170,7 +170,9 @@ export default class DefaultAudioVideoController
       this,
       this._logger
     );
-    this._audioMixController = new DefaultAudioMixController(this._logger);
+    const audioMixController = new DefaultAudioMixController(this._logger);
+    this._audioMixController = audioMixController;
+    this._mediaStreamBroker.addMediaStreamBrokerObserver(audioMixController);
```
Comment (simmkyu) [inline src/audiovideocontroller/DefaultAudioVideoController.ts:175]: Should we also remove the observer in the `destroy` method in the case the customer calls `meetingSession.destory`?

https://github.com/aws/amazon-chime-sdk-js/blob/ae910a680c8d0de920c8bdc639e43f37eb621446/src/meetingsession/DefaultMeetingSession.ts#L135-L137

**Review by simmkyu — CHANGES_REQUESTED** [review]

File: src/audiovideocontroller/DefaultAudioVideoController.ts (hunk 2 above)
Comment (ltrung) [reply src/audiovideocontroller/DefaultAudioVideoController.ts]: Oops. Forget about it. Updated to remove during destroy.

**Review by simmkyu — DISMISSED** [review]

**Review by simmkyu — APPROVED** [review]

**Pull request merged**